File: 2/main.go
package main

import (
	"fmt"
	"math"
)

func main() {

	fmt.Println(Movie(500, 15, 0.9)) // 43
	fmt.Println(Movie(0, 10, 0.95))  // 2
}

// Movie is a func to solve John's problem, can you just solve your problem? f*** you John
func Movie(card, ticket int, perc float64) int {
	counter := 0
	priceA := 0
	priceB := float64(card)

	// prevPrice is a variable to track the progression
	prevPrice := float64(ticket)

	for math.Ceil(priceB) >= float64(priceA) {
		priceA += ticket
		prevPrice *= perc
		priceB += prevPrice
		counter++
	}
	return counter
}

File: 2/main.js
/* function movie(card, ticket, perc) {
    let counter = 0
    let priceA = 0
    let priceB = card

    let prevPrice = ticket

    while (Math.ceil(priceB) >= priceA) {
        priceA += ticket
        prevPrice *= perc
        priceB += prevPrice
        counter++
    }

    return counter
}; */

function movie(card, ticket, perc) {
    for (
        var s=perc, n=1;
        Math.ceil(card+s*ticket) >= ticket*n;
        s+= Math.pow(perc, ++n)
        ){}
    return n
}

console.log(movie(500, 15, 0.9)) //43
console.log(movie(0, 10, 0.95)) //2